xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/internal/macro/AbstractShowHideIfMacro.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.macros.showhideif.internal.macro;
+
+import java.util.Collection;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.macro.MacroContentParser;
+import org.xwiki.rendering.macro.MacroExecutionException;
+import org.xwiki.rendering.macro.descriptor.DefaultContentDescriptor;
+import org.xwiki.user.group.GroupException;
+import org.xwiki.user.group.GroupManager;
+import org.xwiki.user.group.WikiTarget;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xwiki.macros.AbstractProMacro;
+import com.xwiki.macros.internal.grouplist.GroupReferenceList;
+import com.xwiki.macros.internal.userlist.UserReferenceList;
+import com.xwiki.macros.showhideif.macro.MacroParameter;
+
+/**
+ * Base class for macros hide-if and show-if. This is mainly used to define if the constraint match or not.
+ *
+ * @version $Id: $
+ */
+public abstract class AbstractShowHideIfMacro extends AbstractProMacro<MacroParameter>
+{
+    private static final String CONTENT_DESCRIPTION = "the content to show conditionally";
```

Comment:
Since the description will be used in the ckeditor (if no localization is provided), you should format it in a nicer way.